### PR TITLE
docs: add advanced software mgmt descriptions

### PR DESCRIPTION
**Pull request by rina23q**
## Proposed changes

These config keys were not described anywhere, so I would add a short explanation.
```sh
sudo tedge config set c8y.software_management.api advanced
sudo tedge config set c8y.software_management.with_types true
```

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: docs/src/operate/c8y/software-management.md [hunk 1]
```diff
@@ -112,6 +112,19 @@ If you are upgrading %%te%% from any version prior to `1.0.0` (including `1.0.0-
 
 ## Configuration {#configuration}
 
+### Advanced Software Management
```
Comment (didier-wenzek) [inline docs/src/operate/c8y/software-management.md:115]: An introduction is missing to tell what makes this feature *advanced*.

Comment (gligorisaev) [reply docs/src/operate/c8y/software-management.md:115]: something like:?

The Advanced Software Management feature in Cumulocity IoT enhances the traditional software management capabilities by providing more flexible and comprehensive control over software operations on devices. This advanced feature supports detailed software types and more granular control over software management processes, allowing for improved management and monitoring of software across diverse device fleets.

Comment (rina23q) [reply docs/src/operate/c8y/software-management.md:115]: I'm not sure if we should explain what is advanced software management. It's purely c8y's feature and terminology, and for thin-edge, it's just an endpoint difference. Isn't it enough to give a link to the c8y docs (there's a link in the first sentence of this paragraph)

Comment (didier-wenzek) [reply docs/src/operate/c8y/software-management.md:115]: Yes, a link to Cumulocity docs is enough.

Comment (rina23q) [reply docs/src/operate/c8y/software-management.md]: Updated the description [861c9bd](https://github.com/thin-edge/thin-edge.io/pull/2985/commits/861c9bd37ba506fac7ec6c702e98c1da7b08725f)

**Review by didier-wenzek — APPROVED** [review]
Approved

File: docs/src/operate/c8y/software-management.md [hunk 2]
```diff
@@ -112,6 +112,20 @@ If you are upgrading %%te%% from any version prior to `1.0.0` (including `1.0.0-
 
 ## Configuration {#configuration}
 
+### Switching to Advanced Software Management
+
+%%te%% supports both [Legacy Software Management](https://cumulocity.com/docs/device-integration/fragment-library/#legacy-software-management) and [Advanced Software Management](https://cumulocity.com/docs/device-integration/fragment-library/#advanced-software-management) features of Cumulocity IoT.
+By default, Legacy Software Management is selected.
+To switch to Advanced Software Management, use the `tedge` command to change the following configurations:
```
Comment (reubenmiller) [inline docs/src/operate/c8y/software-management.md:119]: We should probably also mention that the mapper needs to be restarted before the setting takes effect.

Comment (rina23q) [reply docs/src/operate/c8y/software-management.md:119]: Good point. Added.

**Review by reubenmiller — APPROVED** [review]
Approved. nice improvement

**Pull request merged**